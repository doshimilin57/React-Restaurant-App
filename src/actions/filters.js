import { SET_TEXT_FILTER } from './types';

// SET_TEXT_FILTER
export const setTextFilter = (text = '') => ({
  type: SET_TEXT_FILTER,
  text,
});
