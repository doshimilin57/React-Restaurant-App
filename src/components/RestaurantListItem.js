import React from 'react';
import numeral from 'numeral';

const RestaurantListItem = ({
  id,
  name,
  price,
  address,
  city,
  state,
  country,
  postal_code,
}) => (
  <div className="list-item">
    <div>
      <h3 className="list-item__title">{name}</h3>
      <span className="list-item__sub-title">
        {address}
        {', '}
        {city}
        {', '}
        {state}
        {', '}
        {country}
        {', '}
        {postal_code}
      </span>
    </div>
    <h3 className="list-item__data">{numeral(price).format('$0,0.00')}</h3>
  </div>
);

export default RestaurantListItem;
