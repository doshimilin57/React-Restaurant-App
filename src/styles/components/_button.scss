.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
  height: 100%;
}

.button--link {
  background: none;
  color: #888;
  font-size: 1.2rem;
}

.button--secondary {
  background: #888;
}
