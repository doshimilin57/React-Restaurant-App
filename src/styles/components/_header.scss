.header {
  background-color: #ba9a8e;
  background-image: linear-gradient(315deg, #ba9a8e 0%, #96705b 74%);
}

.header__content {
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding: $s-size 0;
}

.header__title {
  color: white;
  text-decoration: none;
  h1 {
    margin: 0;
  }
}
