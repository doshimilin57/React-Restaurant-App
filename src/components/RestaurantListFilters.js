import React from "react";
import { connect } from "react-redux";
import { setTextFilter } from "../actions/filters";
import selectRestaurants from "../selectors/restaurants";
import {
  startSearchRestaurants,
  startShowMoreRestaurants,
} from "../actions/restaurants";

export class RestaurantListFilters extends React.Component {
  state = {
    city: "",
    displayRefineBox: false,
  };

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };

  onCityChange = (e) => {
    const city = e.target.value;
    this.setState(() => ({ city }));
  };

  submitHandler = (e) => {
    e.preventDefault();
    this.props.startSearchRestaurants(this.state.city);
  };

  startShowMoreRestaurants = (e) => {
    e.preventDefault();
    this.props.startShowMoreRestaurants(this.state.city);
  };

  render() {
    return (
      <div className="content-container list-filters">
        <form onSubmit={this.submitHandler}>
          <div className="input-group">
            <div className="input-group__item">
              <input
                type="text"
                className="text-input"
                placeholder="Enter City"
                aria-labelledby="lblCity"
                tabindex="-1"
                value={this.state.city}
                onChange={this.onCityChange}
              />
            </div>
            <div className="input-group__item">
              <input
                type="text"
                className="text-input"
                // placeholder="Filter(Name, Address, Area)"
                placeholder="Filter Data"
                aria-labelledby="lblFilter"
                tabindex="-1"
                value={this.props.filters.text}
                onChange={this.onTextChange}
              />
            </div>
            <div>
              <button type="submit" className="button">
                Search
              </button>
            </div>
            <div>
              <button
                type="click"
                className="button button--link"
                onClick={this.startShowMoreRestaurants}
                disabled={
                  this.props.restaurantCount >= this.props.total_entries
                }
              >
                Show More
              </button>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  const visibleRestaurants = selectRestaurants(
    state.restaurants,
    state.filters
  );
  return {
    filters: state.filters,
    total_entries: state.results.total_entries,
    restaurantCount: visibleRestaurants.length,
  };
};

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  startSearchRestaurants: (city) => dispatch(startSearchRestaurants(city)),
  startShowMoreRestaurants: (city) => dispatch(startShowMoreRestaurants(city)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(RestaurantListFilters);
