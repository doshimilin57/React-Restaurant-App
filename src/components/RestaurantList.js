import React from 'react';
import { connect } from 'react-redux';
import RestaurantListItem from './RestaurantListItem';
import selectRestaurants from '../selectors/restaurants';

export const RestaurantList = ({ restaurants }) => (
  <div className="content-container">
    <div className="list-header">
      <div className="show-for-mobile">Restaurants</div>
      <div className="show-for-desktop">Restaurant</div>
      <div className="show-for-desktop">Price</div>
    </div>
    <div className="list-body">
      {restaurants.length === 0 ? (
        <div className="list-item list-item--message">
          <span>No results!</span>
        </div>
      ) : (
        restaurants.map((restaurant) => {
          return <RestaurantListItem key={restaurant.id} {...restaurant} />;
        })
      )}
    </div>
  </div>
);

const mapStateToProps = (state) => {
  return {
    restaurants: selectRestaurants(state.restaurants, state.filters),
  };
};

export default connect(mapStateToProps)(RestaurantList);
