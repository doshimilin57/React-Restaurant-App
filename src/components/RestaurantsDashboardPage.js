import React from "react";
import RestaurantList from "./RestaurantList";
import RestaurantListFilters from "./RestaurantListFilters";

const RestaurantsDashboardPage = () => (
  <div>
    <RestaurantListFilters />
    <RestaurantList />
  </div>
);

export default RestaurantsDashboardPage;
