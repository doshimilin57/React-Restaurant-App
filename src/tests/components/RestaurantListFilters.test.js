import React from 'react';
import { shallow } from 'enzyme';
import { RestaurantListFilters } from '../../components/RestaurantListFilters';
import { filters, altFilters } from '../fixtures/filters';

let setTextFilter, startSearchRestaurants, wrapper;

beforeEach(() => {
  setTextFilter = jest.fn();
  startSearchRestaurants = jest.fn();
  wrapper = shallow(
    <RestaurantListFilters
      setTextFilter={setTextFilter}
      startSearchRestaurants={startSearchRestaurants}
      filters={filters}
      city={'toronto'}
    />
  );
});

test('should render RestaurantListFilters properly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render RestaurantListFilters with altFilters properly', () => {
  wrapper.setProps({ filters: altFilters });
  expect(wrapper).toMatchSnapshot();
});

test('should handle form submission properly', () => {
  const value = 'toronto';
  wrapper.find('input').at(0).simulate('change', { target: { value } });
  wrapper.find('form').simulate('submit', {
    preventDefault: () => {},
  });
  expect(startSearchRestaurants).toHaveBeenLastCalledWith(value);
});

test('should handle city texbox change properly', () => {
  const value = 'toronto';
  wrapper.find('input').at(0).simulate('change', { target: { value } });
  expect(wrapper.state('city')).toBe(value);
});

test('should handle filter textbox change properly', () => {
  const value = 'toronto';
  wrapper.find('input').at(1).simulate('change', { target: { value } });
  expect(setTextFilter).toHaveBeenLastCalledWith(value);
});
