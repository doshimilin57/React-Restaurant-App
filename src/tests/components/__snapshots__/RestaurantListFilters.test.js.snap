// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RestaurantListFilters properly 1`] = `
<div
  className="content-container list-filters"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="input-group"
    >
      <div
        className="input-group__item"
      >
        <input
          aria-labelledby="lblCity"
          className="text-input"
          onChange={[Function]}
          placeholder="Enter City"
          tabindex="-1"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-group__item"
      >
        <input
          aria-labelledby="lblFilter"
          className="text-input"
          onChange={[Function]}
          placeholder="Filter Data"
          tabindex="-1"
          type="text"
          value=""
        />
      </div>
      <div>
        <button
          className="button"
          type="submit"
        >
          Search
        </button>
      </div>
      <div>
        <button
          className="button button--link"
          disabled={false}
          onClick={[Function]}
          type="click"
        >
          Show More
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`should render RestaurantListFilters with altFilters properly 1`] = `
<div
  className="content-container list-filters"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="input-group"
    >
      <div
        className="input-group__item"
      >
        <input
          aria-labelledby="lblCity"
          className="text-input"
          onChange={[Function]}
          placeholder="Enter City"
          tabindex="-1"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-group__item"
      >
        <input
          aria-labelledby="lblFilter"
          className="text-input"
          onChange={[Function]}
          placeholder="Filter Data"
          tabindex="-1"
          type="text"
          value="toronto"
        />
      </div>
      <div>
        <button
          className="button"
          type="submit"
        >
          Search
        </button>
      </div>
      <div>
        <button
          className="button button--link"
          disabled={false}
          onClick={[Function]}
          type="click"
        >
          Show More
        </button>
      </div>
    </div>
  </form>
</div>
`;
